docs/technical-overview/feature_flags.md

```diff
@@ -0,0 +1,87 @@
+# Feature flags
+
+We sometimes employ feature flags to develop features that aren't fully formed
+at the moment of deployment. They can also be employed to roll out features in
+productions incrementally.
+
+Feature flags are meant to be temporary and part of a rollout plan resulting in
+their removal.
+
+## Creating a new feature flag
+
+To create a new feature flag, we use data update scripts so that they can be
+made available over the entire Forem fleet, for example:
+
+```shell
+bundle exec rails g data_update AddAwesomeAlgorithmFeatureFlag
+```
+
+this will generate a script and its spec. Once done, you can add the new feature
+flag like this:
+
+```ruby
+module DataUpdateScripts
+  class AddAwesomeAlgorithmFeatureFlag
+    def run
+      FeatureFlag.add(:awesome_algorithm)
+    end
+  end
+end
+```
+
+and then start using it right away.
+
+## Guarding a feature behind a flag
+
+Once the feature flag is added, you can start using it to hide feature behind a
+boolean flag:
+
+```ruby
+if FeatureFlag.enabled?(:awesome_algorithm)
+  # call the new code
+else
+  # call the previous code
+end
+```
+
+or, for example in a view:
+
+```erb
+<% if FeatureFlag.enabled?(:blue_button) %>
+  <%= render "blue_button" %>
+<% else %>
+  <%= render "good_old_button" %>
+<% end %>
+```
+
+## Enabling/disabling a feature flag globally
+
+As mentioned, feature flags can be used to test a work in progress feature or to
+test a new approach directly in production.
+
+To enable such feature, login to the Forem, go to `/admin/feature_flags` (it
+requires the `tech_admin` role), click on the flag you want to enable and press
+"Fully Enable".
```

Comment:
(non-blocking) This seems a little hard to follow. I think presenting it as a list is easier for me to read. What do you think?

```suggestion
To enable such a feature:
- Login to the Forem
- Go to `/admin/feature_flags` (it requires the `tech_admin` role)
- Click on the flag you want to enable and press "Fully Enable".
``

Reply:
I like the list view, can we change `login` to `log in` and a suggestion is to possibly add the link to the docs on how to add a role.

Reply:
@Ridhwana will do manually!